### PR TITLE
Add CheckBuildDate from openSUSE/rpmlint-checks (#227).

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

File: rpmlint/checks/BuildDateCheck.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        grep_date = pkg.grep(self.istoday, filename)
+
+        if len(grep_date):
+            grep_time = pkg.grep(self.looksliketime, filename)
+
+            if len(grep_time):
+                self.output.add_info('W', pkg, 'file-contains-date-and-time', filename)
+            else:
+                self.output.add_info('W', pkg, 'file-contains-current-date', filename)
```
Comment (scarabeusiv) [inline rpmlint/checks/BuildDateCheck.py]: I would even error out, these when happen are so much pain in the arse for rebuilds...

File: rpmlint/descriptions/CheckBuildDate.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+obsolete-xinetd-requirement="""
+In systemd based distributions xinetd has become obsolete.
+Please remove dependencies on xinetd."""
```
[the remark this answers is not in the document]
Comment (marxin) [reply rpmlint/descriptions/CheckBuildDate.toml]: Whoops.

File: rpmlint/descriptions/CheckBuildDate.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+file-contains-current-date="""
+Your file contains the current date, this may cause the package
+to rebuild in excess."""
+file-contains-date-and-time="""
+Your file uses  __DATE and __TIME__ this causes the package to
```
Comment (scarabeusiv) [inline rpmlint/descriptions/CheckBuildDate.toml]: The variable here should be __DATE__ iirc

File: test/test_lint.py [hunk 4]
```diff
@@ -38,6 +39,18 @@
 ]
 
 
+def parse_err(err):
```
Comment (scarabeusiv) [inline test/test_lint.py]: Nope this all hies issues, we can't be unable to parse stuff, it should gracefully skip stuff and not print errors about encodings.

Comment (marxin) [reply test/test_lint.py]: I see. I've rebased the patch, can you please take a look?

**Comment by scarabeusiv** [comment]
5.77s$ docker run -v $(pwd):/usr/src/rpmlint/ $DIST flake8
./rpmlint/pkg.py:580:9: F841 local variable 'e' is assigned to but never used

**Comment by marxin** [comment]
> 5.77s$ docker run -v $(pwd):/usr/src/rpmlint/ $DIST flake8
> ./rpmlint/pkg.py:580:9: F841 local variable 'e' is assigned to but never used

Fixed.

**Pull request merged**